Add Simulon remote access to docs (#547)
XanaduAI/strawberryfields: doc/introduction/photonic_hardware.rst

Xanadu Quantum Cloud

> ====================

Reviewer: This change makes sense, but at the same time I am hesitant - for a new user, seeing 'Hardware' in the side bar is much more descriptive and enticing than XQC.

Can we preserve 'hardware' somehow?

Author: Sure, "Hardware and cloud" might work? I cannot think of anything better really. 🤔

Using Strawberry Fields, you can submit quantum programs to be executed on photonic

> hardware or to be run on a cloud simulator via the Xanadu cloud platform.

Reviewer: ```suggestion
hardware or to be run on a cloud simulator via the Xanadu Quantum Cloud.
```

After executing the above command, the result will be stored in ``out.txt`` in the current working directory.
You can also omit the ``--output`` parameter to print the result to the screen.

Cloud simulator
---------------

> Instead of submitting a job to be run on quantum hardware, it is also possible to run jobs on
> our cloud simulator Simulon. The process is very similar to running jobs on hardware. You

Reviewer: ```suggestion
In addition to submitting jobs to be run on quantum hardware, it is also possible to run jobs on
the Xanadu Quantum Cloud simulator Simulon. The process is very similar to running jobs on hardware. You
```

will need to configure your account, as described above, and submit a job via the ``RemoteEngine``,

> using `"simulon"` as the target instead of a specific chip:

Reviewer: ```suggestion
using ``"simulon"`` as the target instead of a specific chip:
```

>>> eng = sf.RemoteEngine("simulon")
>>> result = eng.run(prog)

> Alternatively, it's also possible to run jobs asynchronously with `eng.run_async` or by submitting a
> Blackbird script, setting the ``target`` to ``simulon`` in the Blackbird header.

Reviewer: ```suggestion
Simulon jobs can also be submitted asynchronously using ``eng.run_async``, or by submitting a
Blackbird script with the ``target`` set to ``simulon`` in the Blackbird header.
```

> See the `Submitting jobs`_ section above for more details.

Author: I'm thinking that a user might scroll down to read just the "Cloud simulator" section, and not understand that everything is detailed in the section above.

See the `Submitting jobs on hardware`_ section above for more details.

.. note::

    Simulon runs on the ``gaussian`` backend (see :ref:`simulating_your_program`) and thus only supports
    Gaussian operations, including homodyne and heterodyne measurements, as well terminal

>     Fock measurements.

Reviewer 2: Nice! 🙂 Could it be worth adding just a small comment that the number of modes depends on the type & number of measurements used? Without stating anything explicitly.
